lib/medtronic/medtronicSimulator.js

```diff
@@ -413,23 +440,29 @@ exports.make = function(config){
       // hasn't already been triggered
       if(suspendingEvent === null) {
         setSuspendingEvent(event);
+      } else {
+        simpleSimulate(event);
       }
-      simpleSimulate(event);
     },
     changeDeviceTime: function(event) {
       simpleSimulate(event);
     },
     finalBasal: function(finalRecordTime) {
```

Comment:
Don't you have the tools at this point to check against the active schedule to guess at the final basal's duration, or am I missing something?